src/plugins/intel_npu/src/plugin/npuw/serialization.cpp

```diff
@@ -0,0 +1,184 @@
+    std::size_t var_size = 0;
+    stream.read(reinterpret_cast<char*>(&var_size), sizeof var_size);
+    var.resize(var_size);
+    stream.read(&var[0], var_size);
+}
+
+void ov::npuw::s11n::read(std::istream& stream, bool& var) {
+    stream.read(reinterpret_cast<char*>(&var), sizeof var);
+}
+
+void ov::npuw::s11n::read(std::istream& stream, ov::npuw::compiled::Spatial& var) {
+    using ov::npuw::s11n::read;
+
+    ov::npuw::compiled::Spatial spat;
+    std::size_t params_size = 0;
+    read(stream, params_size);
+    for (std::size_t i = 0; i < params_size; ++i) {
+        ov::npuw::compiled::Spatial::Param p;
+        read(stream, p.idx);
+        read(stream, p.dim);
+        spat.params.push_back(p);
+    }
+    read(stream, spat.range);
+    read(stream, spat.nway);
+    read(stream, spat.out_dim);
+    read(stream, spat.nway_iters);
+    read(stream, spat.tail_size);
+}
+
+void ov::npuw::s11n::read(std::istream& stream, ov::Tensor& var) {
+    std::string type_str;
+    read(stream, type_str);
+    ov::element::Type type(type_str);
+
+    ov::Shape shape;
+    read(stream, shape);
+
+    std::size_t byte_size = 0;
+    read(stream, byte_size);
+
+    var = ov::Tensor(type, shape);
+
+    stream.read(reinterpret_cast<char*>(var.data()), byte_size);
+}
+
+void ov::npuw::s11n::read(std::istream& stream, ::intel_npu::Config& var) {
+    std::string str_cfg;
+    read(stream, str_cfg);
+
+    // Parse cfg from string
+    std::map<std::string, std::string> config;
+
+    size_t pos = 0;
+    std::string token, key, value;
+    while ((pos = str_cfg.find(' ')) != std::string::npos) {
+        token = str_cfg.substr(0, pos);
+        auto pos_eq = token.find('=');
+        key = token.substr(0, pos_eq);
+        value = token.substr(pos_eq + 2, token.size() - pos_eq - 3);
+        config[key] = value;
+        str_cfg.erase(0, pos + 1);
+    }
+
+    // Process tail
+    auto pos_eq = str_cfg.find('=');
+    key = str_cfg.substr(0, pos_eq);
+    value = str_cfg.substr(pos_eq + 2, str_cfg.size() - pos_eq - 3);
+    config[key] = value;
+
+    var.update(config);
+}
+
+void ov::npuw::s11n::read(std::istream& stream, std::vector<std::shared_ptr<ov::op::v0::Parameter>>& var) {
+    var.clear();
+    std::size_t params_size = 0;
+    read(stream, params_size);
+    for (std::size_t i = 0; i < params_size; ++i) {
+        std::string elem_type_str;
+        std::string part_shape_str;
+        std::unordered_set<std::string> names;
+        read(stream, elem_type_str);
+        read(stream, part_shape_str);
+        read(stream, names);
+        // NOTE: the code below is taken from NPU plugin's create_dummy_model()
+        auto param =
+            std::make_shared<op::v0::Parameter>(ov::element::Type(elem_type_str), ov::PartialShape(part_shape_str));
+        param->set_friendly_name(*names.begin());  // FIXME: any_name ?
+        param->output(0).get_tensor().set_names(names);
+        var.push_back(param);
+    }
+}
```

Comment:
shouldn't this generalize via `read(vector<T>)`, if you already have one?

Reply:
..so you'd only need to define `read(shared_ptr<Parameter>&)`

Reply:
Done